File: src/test/scala/package.scala
/* -*-mode:scala; c-basic-offset:2; indent-tabs-mode:nil -*- */
package com.jcraft.jsch

import scala.language.implicitConversions
import scala.language.postfixOps
import scala.language.reflectiveCalls

import java.io._

package object jzlib {
  implicit def readIS(is: InputStream) = new {
      def ->(out: OutputStream)
            (implicit buf: Array[Byte] = new Array[Byte](1024)) = {
      LazyList.
        continually(is.read(buf)).
        takeWhile(-1 !=).
        foreach(i => out.write(buf, 0, i))
      is.close
    }
  }

  // reading a resource file 
  implicit def fromResource(str: String ) = new {
    def fromResource: Array[Byte] =
      io.Source.
         fromURL(getClass.getResource(str))(io.Codec.ISO8859).
         map(_.toByte).
         toArray
  }

  implicit def readArray(is: Array[Byte]) = new {
      def ->(out: OutputStream)(implicit buf: Array[Byte]) = {
        new ByteArrayInputStream(is) -> (out)
    }
  }

  def randombuf(n: Int) = (0 to n).map{ _ =>
    util.Random.nextLong().asInstanceOf[Byte] 
  }.toArray
}

File: src/test/java/com/jcraft/jsch/jzlib/Package.java
package com.jcraft.jsch.jzlib;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.io.UncheckedIOException;
import java.util.Random;
import java.util.function.Consumer;
import java.util.function.Function;

public class Package {
  public static void readIS(InputStream is, OutputStream out, byte[] buf) throws IOException {
    int i;
    while ((i = is.read(buf)) != -1) {
      out.write(buf, 0, i);
    }
    is.close();
  }

  public static void readArray(byte[] is, OutputStream out, byte[] buf) throws IOException {
    readIS(new ByteArrayInputStream(is), out, buf);
  }

  public static byte[] randombuf(int n) {
    Random random = new Random();
    byte[] ret = new byte[n];
    random.nextBytes(ret);
    return ret;
  }

  public static <T> Consumer<T> uncheckedConsumer(IOConsumer<T> consumer) {
    return t -> {
      try {
        consumer.accept(t);
      } catch (IOException e) {
        throw new UncheckedIOException(e);
      }
    };
  }

  public static <T, R> Function<T, R> uncheckedFunction(IOFunction<T, R> function) {
    return t -> {
      try {
        return function.apply(t);
      } catch (IOException e) {
        throw new UncheckedIOException(e);
      }
    };
  }

  @FunctionalInterface
  public interface IOConsumer<T> {
    void accept(T t) throws IOException;
  }

  @FunctionalInterface
  public interface IOFunction<T, R> {
    R apply(T t) throws IOException;
  }
}
